source/code/plugins/tailfilereader.rb

```diff
@@ -332,31 +360,37 @@ class FilePositionEntry
       LN_OFFSET = 33
       SIZE = 34
 
-      def initialize(file, seek)
+      def initialize(file, file_mutex, seek, pos, inode)
         @file = file
+        @file_mutex = file_mutex
         @seek = seek
+        @pos = pos
+        @inode = inode
       end
 
       def update(ino, pos)
-        @file.pos = @seek
-        @file.write "%016x\t%016x" % [pos, ino]
+        @file_mutex.synchronize {
+          @file.pos = @seek
+          @file.write "%016x\t%016x" % [pos, ino]
+        }
+        @pos = pos
+        @inode = ino
       end
 
       def update_pos(pos)
-        @file.pos = @seek
-        @file.write "%016x" % pos
+        @file_mutex.synchronize {
+          @file.pos = @seek
+          @file.write "%016x" % pos
+        }
+        @pos = pos
       end
 
       def read_inode
-        @file.pos = @seek + INO_OFFSET
-        raw = @file.read(INO_SIZE)
-        raw ? raw.to_i(16) : 0
+        @inode
```

Comment:
we removed this and below?

Reply:
how are we going to verify these changes? will our test automation suite cover these scenarios?

Reply:
also, we need to make identical change in DSC repo as these are shipped as part of nxOMSSudoCustomLog module

Reply:
So instead of reading from file every time and calculating the inode and pos within this method, we pass in these two values to the object when we initialize it and store it in the instance variables, which are updated whenever we call the update method. So it gets rid of the redundant file reads. 
Position File entry is crucial in the tailing process and if it wasn't working correctly the logs wouldn't be tailed correctly, which is tested by the unit tests and the manual testing.
I will create a DSC repo PR once this is merged.